extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/DbCheckingPMetaSupplier.java

```diff
@@ -0,0 +1,195 @@
+        return forceTotalParallelismOne ? 1 : preferredLocalParallelism;
+    }
+
+    @Nullable
+    @Override
+    public Permission getRequiredPermission() {
+        return requiredPermission;
+    }
+
+    /**
+     * If true, only one instance of given supplier will be created.
+     */
+    public DbCheckingPMetaSupplier forceTotalParallelismOne(boolean forceTotalParallelismOne) {
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        return this;
+    }
+
+    @Override
+    public boolean initIsCooperative() {
+        return true;
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        if (forceTotalParallelismOne) {
+            preferredLocalParallelism = 1;
+            if (context.localParallelism() != 1) {
+                throw new IllegalArgumentException(
+                        "Local parallelism of " + context.localParallelism() + " was requested for a vertex that "
+                                + "supports only total parallelism of 1. Local parallelism must be 1.");
+            }
+            String key = getPartitionKey(newUnsecureUuidString());
+            int partitionId = context.hazelcastInstance().getPartitionService().getPartition(key).getPartitionId();
+            ownerAddress = context.partitionAssignment().entrySet().stream()
+                                  .filter(en -> arrayIndexOf(partitionId, en.getValue()) >= 0)
+                                  .findAny()
+                                  .map(Entry::getKey)
+                                  .orElseThrow(() -> new RuntimeException("Owner partition not assigned to any " +
+                                          "participating member"));
+        }
+
+        if (shouldCheck) {
+            Tuple2<MongoClient, DataConnection> clientAndRef = connect(context);
+            MongoClient client = clientAndRef.requiredF0();
+            try {
+                if (databaseName != null) {
+                    checkDatabaseExists(client, databaseName);
+                    MongoDatabase database = client.getDatabase(databaseName);
+                    if (collectionName != null) {
+                        checkCollectionExists(database, collectionName);
+                    }
+                }
+            } finally {
+                DataConnection connection = clientAndRef.f1();
+                if (connection != null) {
+                    connection.release();
+                }
+            }
+        }
+    }
+
+    private Tuple2<MongoClient, DataConnection> connect(Context context) {
+        try {
+            if (clientSupplier != null) {
+                return tuple2(clientSupplier.get(), null);
+            } else if (dataConnectionRef != null) {
+                NodeEngineImpl nodeEngine = Util.getNodeEngine(context.hazelcastInstance());
+                InternalDataConnectionService dataConnectionService = nodeEngine.getDataConnectionService();
```

Comment:
you can use public API here (`DataConnectionService` and `NodeEngine`)

Reply:
66a12252c410af396652a53af17c5bdf48630dda